### PR TITLE
[Merged by Bors] - chore: Make `Set`/`Finset` lemmas match lattice lemma names

### PR DESCRIPTION
**Pull request by YaelDillies**
Rename `union_eq_left_iff_subset` to `union_eq_left` to match `sup_eq_left`. Similarly for the `right` and `inter` versions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
These renames look reasonable, thanks! Thanks for not renaming too many lemmas at once.

Since this is somewhat bike-sheddy, I'll let someone else make the final call.

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors d+

**Comment by YaelDillies** [comment]
bors merge

File: Mathlib/Data/Set/Basic.lean [hunk 1]
```diff
@@ -804,21 +804,21 @@ theorem union_right_comm (s₁ s₂ s₃ : Set α) : s₁ ∪ s₂ ∪ s₃ = s
...
   sup_eq_left
-#align set.union_eq_left_iff_subset Set.union_eq_left_iff_subset
+#align set.union_eq_left Set.union_eq_left
 
 @[simp]
-theorem union_eq_right_iff_subset {s t : Set α} : s ∪ t = t ↔ s ⊆ t :=
+theorem union_eq_right {s t : Set α} : s ∪ t = t ↔ s ⊆ t :=
   sup_eq_right
-#align set.union_eq_right_iff_subset Set.union_eq_right_iff_subset
+#align set.union_eq_right Set.union_eq_right
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Basic.lean:814]: Note that this incorrectly changed the lean3 name, but #7529 fixed it.

Comment (YaelDillies) [reply Mathlib/Data/Set/Basic.lean:814]: I tried to pay attention not to do that, but apparently not enough 😦

**Pull request closed without merging**